packages/node_modules/samples/browser-auth-implicit/app.js

```diff
@@ -4,8 +4,29 @@
 /* eslint-disable no-console */
 /* eslint-disable require-jsdoc */
 
-// Declare some globals that we'll need throughout
-const spark = ciscospark.init();
+
+// This is one way you might programattically determine your redirect_uri
+// depending on where you've deployed your app, but you're probably better off
+// having development/staging/production builds and injecting directly from the
+// environment.
+let redirect_uri = `${window.location.protocol}//${window.location.host}`;
+if (window.location.pathname) {
+  redirect_uri += window.location.pathname;
+}
+
+// Inititate the SDK
+// IN A PRODUCTION APP YOU SHOULD NOT HARDCODE THESE VALUES BUT INSTEAD LOAD
+// THEM FROM THE ENVIRONMENT AT BUILD TIME. SECRETS AND OTHER CREDENTIALS SHOULD
+// NOT BE COMMITTED TO THE CODEBASE.
+const spark = ciscospark.init({
+  config: {
+    credentials: {
+      client_id: `C7c3f1143a552d88d40b2afff87600c366c830850231597fb6c1c1e28a5110a4f`,
```

Comment:
Given the warning, does it make sense to hard code it here, or provide some instructions on how to add it into envvars?

Reply:
I thought about that. The problem is that none of the sample apps are compiled, they're loaded as-is. My goal was to have as little friction as possible to interpret the sample code. 

I guess an alternative might be to put a form at the top of the page for them to add their app credentials, but then the sample would get a whole lot bigger.